tracer/test/test-applications/integrations/Samples.SqlServer/StoredProcedures.cs

```diff
@@ -0,0 +1,267 @@
+using System;
+using System.Data;
+using System.Data.SqlClient;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Transactions;
+
+namespace Samples.SqlServer
+{
+    public class StoredProcedure
+    {
+        public static async Task RunStoredProcedureTestAsync(string connectionString, CancellationToken token)
+        {
+            using var connection = new SqlConnection(connectionString += ";Pooling=false");
+            await connection.OpenAsync(token);
+            Console.WriteLine("Starting SQL Server Stored Procedure Calls");
+            // all the SQL here is based on DbCommandFactory
+
+            var tableName = $"[Stored-Proc-System-Data-SqlClient-Test-{Guid.NewGuid():N}]";
+
+            using (var command = connection.CreateCommand())
+            {
+                command.CommandText = $"CREATE TABLE {tableName} (Id int PRIMARY KEY, Name varchar(100));";
+                await command.ExecuteNonQueryAsync(token);
+
+                command.CommandText = $"INSERT INTO {tableName} (Id, Name) VALUES (1, 'Name1'), (2, 'Name2'), (3, 'Name3');";
+                await command.ExecuteNonQueryAsync(token);
+            }
+
+            // Run various stored procedure tests
+            try
+            {
+                await TestBasicStoredProc(connection, tableName, token); // just a basic query
+                await TestOutputParameters(connection, tableName, token); // test OUTPUT and RETURN
+                await TestTransactionScope(connection, tableName, token); // we had exceptions in transaction scopes before so ensure we handle these
+            }
+            catch (Exception)
+            {
+                throw;
+            }
+            finally
+            {
+                // Clean up
+                using (var command = connection.CreateCommand())
+                {
+                    command.CommandText = $@"
+IF OBJECT_ID('dbo.sp_GetTableRow', 'P') IS NOT NULL DROP PROCEDURE dbo.sp_GetTableRow;
+IF OBJECT_ID('dbo.sp_UpdateRowWithOutput', 'P') IS NOT NULL DROP PROCEDURE dbo.sp_UpdateRowWithOutput;
+IF OBJECT_ID('dbo.sp_BatchUpdate', 'P') IS NOT NULL DROP PROCEDURE dbo.sp_BatchUpdate;
+DROP TABLE {tableName};";
+                    await command.ExecuteNonQueryAsync(token);
+                }
+            }
+
+            connection.Close();
+            SqlConnection.ClearAllPools(); // clear the connection pool to avoid issues with lingering connections
+        }
+
+        private static async Task TestBasicStoredProc(SqlConnection connection, string tableName, CancellationToken token)
+        {
+            // Similar to DbCommandFactory.GetSelectRowCommand
+            Console.WriteLine("Test dbo.sp_GetTableRow");
+            using (var command = connection.CreateCommand())
+            {
+                command.CommandText = $@"
```

Comment:
tiny tiny tiny nit: `"""` is awesome and should be your go-to 😄

Reply:
Yep totally forgot that was a thing thanks!